OpenEdXMobile/src/main/java/org/edx/mobile/view/FormFieldActivity.java

```diff
@@ -30,6 +35,26 @@ public static Intent newIntent(@NonNull Context context, @NonNull FormField fiel
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         blockDrawerFromOpening();
+
+        switch (field.getFieldType()) {
+            case SELECT: {
+                final String screenName;
+                final DataType dataType = field.getDataType();
+                if (dataType == COUNTRY) {
+                    screenName = Analytics.Screens.PROFILE_CHOOSE_LOCATION;
+                } else if (dataType == LANGUAGE) {
+                    screenName = Analytics.Screens.PROFILE_CHOOSE_LANGUAGE;
+                } else {
+                    screenName = Analytics.Screens.PROFILE_CHOOSE_BIRTH_YEAR;
+                }
+                environment.getAnalyticsRegistry().trackScreenView(screenName);
+                break;
+            }
```

Comment:
There could be a separate method to get screen name against dataType to keep case statement area smaller and more readable.